main.go

```diff
@@ -70,37 +78,193 @@ func (fw *flushWriter) Write(p []byte) (n int, err error) {
 	return
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
+/* container for the request variables */
+type RequestVars struct {
+	mission int
+	channels []int
+	format string
+	start int
+	duration int
+}
+
+/* local paths to audio files for all channels belonging to particular time slice */
+type TimeSlice struct {
+	start int
+	end int
+	location []string
+}
+
+type DatabaseVars struct {
+	DB_HOST string `json:"DB_HOST"`
+	DB_PORT int `json:"DB_PORT"`
+	DB_USER string `json:"DB_USER"`
+	DB_PASSWORD string `json:"DB_PASSWORD"`
+	DB_NAME string `json:"DB_NAME"`
+}
+
+func parseParameters(r *http.Request) RequestVars {
+	r.ParseForm()
+
+	var rv RequestVars
+
+	tempMission, err := strconv.Atoi(r.Form["mission"][0])
+	check(err)
+	rv.mission = tempMission
+
+	var tempChannels []int
+	for n := range r.Form["channel"] {
+		ch, err := strconv.Atoi(r.Form["channel"][n])
+		check(err)
+		tempChannels = append(tempChannels, ch)
+	}
+	rv.channels = tempChannels
+
+	tempFormat := r.Form["format"][0]
+	rv.format = tempFormat
+
+	tempStart, err := strconv.Atoi(r.Form["t"][0])
+	check(err)
+	rv.start = tempStart
+
+	tempDuration, err := strconv.Atoi(r.Form["len"][0])
+	check(err)
+	rv.duration = tempDuration
+
+	return rv
+}
+
+func getLocations(rv RequestVars) []TimeSlice {
+
+	var slices []TimeSlice
+
+	dbjson, err := ioutil.ReadFile("./config.json")
+	check(err)
+
+	var dbvars DatabaseVars
+	err = json.Unmarshal(dbjson, &dbvars)
+	check(err)
+
+	dbcreds := dbvars
+	dbinfo := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", dbcreds.DB_HOST, dbcreds.DB_PORT, dbcreds.DB_USER, dbcreds.DB_PASSWORD, dbcreds.DB_NAME)
+	db, err := sql.Open("postgres", dbinfo)
+	check(err)
+	defer db.Close()
+
+	stmt, err := db.Prepare("CREATE TABLE chunks_a AS SELECT * FROM channel_chunks WHERE met_end > $1")
```

Comment:
Temporary tables to get all the audio clips that fall in the requested time. First one gets all chunks that end after the start time, thus eliminating all chunks to the left of the request, then of the remaining chunks those that end to the right of the request. Picture related.
![img_20151219_163142647](https://cloud.githubusercontent.com/assets/3846455/11915498/8faafbaa-a66e-11e5-84b9-a09bf07c48ae.jpg)
We could probably do this without creating tables, but my relational algebra game is weak so halp.